docs/docs/migration-guide.mdx

```diff
@@ -24,6 +26,904 @@ If you still have training data in Markdown format then the recommended approach
+constructor but instead returned by the static method `get_default_config`:
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-policy-default-config" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+
+      def __init__(key1: Text = "value1") -> None:
+          ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python
+  class MyPolicy(Policy):
+
+      def __init__(self, config: Dict[Text, Any]) -> None:
+          ...
+
+      @staticmethod
+      def get_default_config() -> Dict[Text, Any]:
+          return {"key1": "value1"}
+  ```
+
+  </TabItem>
+</Tabs>
+
+**Using End-To-End Features in a Policy**
+
+If you were using a custom [end-to-end policy](stories.mdx#end-to-end-training) in Rasa
+Open Source 2, you had to use the `interpreter` parameter to featurize the tracker
+events manually. If your policy requires end-to-end features in Rasa Open Source 3
+you need to [register](custom-graph-components.mdx#registering-graph-components-with-the-model-configuration)
+your policy with type `ComponentType.POLICY_WITH_END_TO_END_SUPPORT`. The features
+will be precomputed and passed into your policy during training and inference.
+
+:::caution
+End-To-End features will only be computed and provided to your policy if your training
+data actually contains [end-to-end training data](stories.mdx#end-to-end-training).
+:::
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-policy-end-to-end-features" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+      def train(
+          self,
+          training_trackers: List[TrackerWithCachedStates],
+          domain: Domain,
+          interpreter: NaturalLanguageInterpreter,
+          **kwargs: Any,
+      ) -> None:
+          ...
+
+      def predict_action_probabilities(
+          self,
+          tracker: DialogueStateTracker,
+          domain: Domain,
+          interpreter: NaturalLanguageInterpreter,
+          **kwargs: Any,
+      ) -> PolicyPrediction:
+          ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python {2,9,17}
+  @DefaultV1Recipe.register(
+      DefaultV1Recipe.ComponentType.POLICY_WITH_END_TO_END_SUPPORT, is_trainable=True
+  )
+  class MyPolicy(Policy):
+      def train(
+          self,
+          training_trackers: List[TrackerWithCachedStates],
+          domain: Domain,
+          precomputations: Optional[MessageContainerForCoreFeaturization] = None,
+      ) -> Resource:
+          ...
+
+      def predict_action_probabilities(
+          self,
+          tracker: DialogueStateTracker,
+          domain: Domain,
+          precomputations: Optional[MessageContainerForCoreFeaturization] = None,
+          rule_only_data: Optional[Dict[Text, Any]] = None,
+          **kwargs: Any,
+      ) -> PolicyPrediction:
+          ...
```

Comment:
Could illustrate getting the features from the container (and with interpreter in the old example).

Reply:
gave it a shot - it's a bit difficult as this somewhat very internal